File: solutions/1231-replace-elements-with-greatest-element-on-right-side/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* replaceElements(int* arr, int arrSize, int* returnSize) {
    *returnSize=arrSize;
    for(int i=arrSize-1;i>0;i--)
        if(arr[i]>arr[i-1])
            arr[i-1]=arr[i];
    for(int i=1;i<arrSize;i++)
        arr[i-1]=arr[i];
    arr[arrSize-1]=-1;
    return arr;
}

File: solutions/1231-replace-elements-with-greatest-element-on-right-side/solution.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    for(let i=arr.length-1;i>0;i--)
            if(arr[i]>arr[i-1])
                arr[i-1]=arr[i];
        for(let i=1;i<arr.length;i++)
            arr[i-1]=arr[i];
        arr[arr.length-1]=-1;
        return arr;
};
